bson/src/main/org/bson/codecs/configuration/CodecRegistry.java

```diff
@@ -26,9 +26,11 @@
  * {@code Object.equals}. It is not necessary to do so, and the simplest course of action is to rely on Object's implementation, but the
  * implementer may wish to implement a "value comparison" in place of the default "reference comparison."</p>
  *
+ * <p>As of the 4.0 release, this class extends the {@code CodecProvider} interface.  This capability was introduced to enable nesting
```

Comment:
Suggestion: Add a default implementation of the codec provider get method in the interface.

Reply:
I don't think it's possible to provide a correct default implementation. The point of the CodecProvider's get method is so that the CodecRegistry that is passed to it is in turn passed on to any Codecs that may need to look up other Codecs (e.g DocumentCodec or PojoCodec).